libcaf_net/caf/net/ssl/connection.hpp

```diff
@@ -56,6 +56,12 @@ class CAF_NET_EXPORT connection {
   /// @param ret The (negative) result from the preceding call.
   std::string last_error_string(ptrdiff_t ret) const;
 
+  // -- SNI support -----------------------------------------------------------
+
+  /// Sets the SNI hostname for this connection before handshake.
+  /// Must be called before connect().
+  bool set_sni_hostname(const char* hostname) noexcept;
```

Comment:
```suggestion
  [[nodiscard]] bool sni_hostname(const char* hostname) noexcept;
```

Also: no getter on the connection object?